forester/src/epoch_manager.rs

```diff
@@ -799,12 +801,43 @@ impl<R: RpcConnection, I: Indexer<R>> EpochManager<R, I> {
             &self.config.payer_keypair.pubkey(),
             epoch_info.epoch.epoch,
         );
-        rpc.create_and_send_transaction(
```

Comment:
would it make sense to do an additional sanity check whether work is already reported by fetching `forester_epoch_pda` and checking its `has_reported_work` field?